src/System.Device.Gpio/System/Device/Gpio/ExtendedPinMode.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// Platform-dependent description of an alternate pin mode (i.e. a pin that is used for I2C or SPI transfer).
+    /// Depending on the hardware in use, the use will reflect the effective mode ("PWM", "SPI") or the state of the
+    /// mode multiplexer of the CPU ("ALT0", "ALT1").
+    /// </summary>
+    public class ExtendedPinMode : IEquatable<ExtendedPinMode>
+    {
+        private readonly string _modeName;
+
+        /// <summary>
+        /// The mode is not known
+        /// </summary>
+        public static readonly ExtendedPinMode Unknown = new ExtendedPinMode("Unknown");
```

Comment:
for unknown just create an internal class UnknownExtendedPinMode with single instance

Reply:
I need this to be public, since clients (in this case the board class) needs to compare against it.

Reply:
This still can be public, UnknownExtendedPinMode  would be internal and ExtendedPinMode decoupled from `string`